library/src/main/java/com/datatorrent/lib/transform/TransformOperator.java

```diff
@@ -0,0 +1,172 @@
+import com.datatorrent.api.DefaultInputPort;
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.api.Operator;
+import com.datatorrent.api.annotation.InputPortFieldAnnotation;
+import com.datatorrent.api.annotation.OutputPortFieldAnnotation;
+import com.datatorrent.common.util.BaseOperator;
+import com.datatorrent.lib.expression.Expression;
+import com.datatorrent.lib.util.PojoUtils;
+
+/**
+ * This operator can transform given POJO using provided expressions and
+ * return a final POJO as a return of transformation process.
+ *
+ * Following are the mandatory fields that needs to be set for TransformOperator to work:
+ * <ul>
+ *   <li><b>expressionMap</b> : Set how the transformation should happen</li>
+ *   <li><b>inputPort.attr.TUPLE_CLASS</b>: Set class type at input port</li>
+ *   <li><b>outputPort.attr.TUPLE_CLASS</b> : Set class type at output port</li>
+ * </ul>
+ *
+ * The operator uses interaction via {@link Expression} and {@link PojoUtils} to transform given POJO.
+ */
+public class TransformOperator extends BaseOperator implements Operator.ActivationListener
+{
+  private Map<String, String> expressionMap = new HashMap<>();
+  private List<String> expressionImports = new LinkedList<>();
+
+  private transient Map<PojoUtils.Setter, Expression> transformationMap = new HashMap<>();
+  private Class<?> inputClass;
+  private Class<?> outputClass;
+
+  public TransformOperator()
+  {
+    expressionImports.add("java.lang.Math.*");
+    expressionImports.add("org.apache.commons.lang3.StringUtils.*");
+  }
+
+  @InputPortFieldAnnotation(schemaRequired = true) public final transient DefaultInputPort<Object> input = new DefaultInputPort<Object>()
+  {
+    @Override public void setup(Context.PortContext context)
+    {
+      inputClass = context.getValue(Context.PortContext.TUPLE_CLASS);
+    }
+
+    @Override public void process(Object o)
+    {
+      processTuple(o);
+    }
+  };
+
+  @OutputPortFieldAnnotation(schemaRequired = true) public final transient DefaultOutputPort<Object> output = new DefaultOutputPort<Object>()
+  {
+    @Override public void setup(Context.PortContext context)
+    {
+      outputClass = context.getValue(Context.PortContext.TUPLE_CLASS);
+    }
+  };
+
+  private void processTuple(Object in)
+  {
+    if (!inputClass.isAssignableFrom(in.getClass())) {
+      throw new RuntimeException(
+          "Unexpected tuple received. Received class: " + in.getClass() + ". Expected class: " + inputClass.getClass());
+    }
+
+    Object out;
+    try {
+      out = outputClass.newInstance();
+    } catch (InstantiationException | IllegalAccessException e) {
+      throw new RuntimeException("Failed to create new object", e);
+    }
+
+    for (Map.Entry<PojoUtils.Setter, Expression> entry : transformationMap.entrySet()) {
+      PojoUtils.Setter set = entry.getKey();
+      Expression expr = entry.getValue();
+      set.set(out, expr.execute(in));
+    }
+
+    output.emit(out);
```

Comment:
I can't think about any. This is because number of incoming tuples and outgoing tuples are same. There is no change that is happening. There is no question of failure and passing also because if expression compiled then its very unlike to fail.

But please suggest if there is any possibility of AutoMetrics here.